tests/_support/Test/DataType/AbstractDataTypeTestCase.php

```diff
@@ -710,6 +710,18 @@ public function testObjects()
         $this->testDataHelper->assertObjects($this->testObject, 'objects', $this->seed);
     }
 
+    public function testReverseRelation()
+    {
+        // // this will create a couple of objects which can be used for references
+        $testObjects = TestHelper::createEmptyObjects();
+
+        $this->createTestObject('objects');
+
+        $this->refreshObject();
+        $this->assertCount(1, $testObjects[0]->getNonowner());
```

Comment:
can't see where the relation got populated

Reply:
I also struggle a bit with the tests ;-)
But as the reverse relation has the same class as itself as `ownerClass` and the field `objects` and owner field, it should be enough to assign the objects to the `objects` field (this is done in the test by `$this->createTestObject('objects')`. Or am I wrong?

Reply:
The `refreshObject()` method load the object to the testObject property. So this could work:
```suggestion
        $this->assertCount(1, $this->testObject->getNonowner());
```

Reply:
The `$testObjects = TestHelper::createEmptyObjects();` can be removed. The `$this->createTestObject('objects');` creates also an object

Reply:
Oh I'm wrong. The `$this->testObject` should relates to the `$testObjects`. This is missing. You need to give an array with fields to the createTestObject method.